cmd/abapEnvironmentAssemblePackages.go

```diff
@@ -0,0 +1,638 @@
+	response, err := conn.Client.SendRequest("HEAD", conn.Baseurl, nil, conn.Header, nil)
+	if err != nil {
+		if response == nil {
+			return errors.Wrap(err, "Fetching X-CSRF-Token failed")
+		}
+		defer response.Body.Close()
+		errorbody, _ := ioutil.ReadAll(response.Body)
+		return errors.Wrapf(err, "Fetching X-CSRF-Token failed: %v", string(errorbody))
+
+	}
+	defer response.Body.Close()
+	token := response.Header.Get("X-CSRF-Token")
+	conn.Header["X-CSRF-Token"] = []string{token}
+	return nil
+}
+
+func (conn connector) get(appendum string) ([]byte, error) {
+	url := conn.Baseurl + appendum
+	response, err := conn.Client.SendRequest("GET", url, nil, conn.Header, nil)
+	if err != nil {
+		if response == nil {
+			return nil, errors.Wrap(err, "Get failed")
+		}
+		defer response.Body.Close()
+		errorbody, _ := ioutil.ReadAll(response.Body)
+		return errorbody, errors.Wrapf(err, "Get failed: %v", string(errorbody))
+
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	return body, err
+}
+
+func (conn connector) post(importBody string) ([]byte, error) {
+	url := conn.Baseurl + "/builds"
+	response, err := conn.Client.SendRequest("POST", url, bytes.NewBuffer([]byte(importBody)), conn.Header, nil)
+	if err != nil {
+		if response == nil {
+			return nil, errors.Wrap(err, "Post failed")
+		}
+		defer response.Body.Close()
+		errorbody, _ := ioutil.ReadAll(response.Body)
+		return errorbody, errors.Wrapf(err, "Post failed: %v", string(errorbody))
+
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	return body, err
+}
+
+func (conn connector) download(appendum string, downloadPath string) error {
+	url := conn.Baseurl + appendum
+	err := conn.DownloadClient.DownloadFile(url, downloadPath, nil, nil)
+	return err
+}
+
+// ******** BUILD logic ********
+
+func (b *build) start(phase string, inputValues values) error {
+	if err := b.getToken(); err != nil {
+		return err
+	}
+	importBody := inputForPost{
+		phase:  phase,
+		values: inputValues,
+	}.String()
+
+	body, err := b.connector.post(importBody)
+	if err != nil {
+		return err
+	}
+
+	var jBuild jsonBuild
+	json.Unmarshal(body, &jBuild)
+	b.BuildID = jBuild.Build.BuildID
+	b.RunState = jBuild.Build.RunState
+	b.ResultState = jBuild.Build.ResultState
+	b.Phase = jBuild.Build.Phase
+	b.Entitytype = jBuild.Build.Entitytype
+	b.Startedby = jBuild.Build.Startedby
+	b.StartedAt = jBuild.Build.StartedAt
+	b.FinishedAt = jBuild.Build.FinishedAt
+	return nil
+}
+
+func (b *build) get() error {
+	appendum := "/builds('" + b.BuildID + "')"
+	body, err := b.connector.get(appendum)
+	if err != nil {
+		return err
+	}
+	var jBuild jsonBuild
+	json.Unmarshal(body, &jBuild)
+	b.RunState = jBuild.Build.RunState
+	b.ResultState = jBuild.Build.ResultState
+	b.Phase = jBuild.Build.Phase
+	b.Entitytype = jBuild.Build.Entitytype
+	b.Startedby = jBuild.Build.Startedby
+	b.StartedAt = jBuild.Build.StartedAt
+	b.FinishedAt = jBuild.Build.FinishedAt
```

Comment:
same as above...

Reply:
Similar as above. If you would call this for example with a not existing BuildID 'body, err := b.connector.get(appendum)' will thrown an error and returns.